### PR TITLE
Silences UTF16 encoding warning 

### PR DESCRIPTION
**Pull request by halfak**
...by filtering chars before passing words to enchant.

This change is meant to address See https://github.com/rfk/pyenchant/issues/58

I didn't see a clear place to add a test, so here's a demo of the fix on the CLI.  

**$ python3**
```
Python 3.4.0 (default, Apr 11 2014, 13:05:11) 
[GCC 4.8.2] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import enchant
>>> d = enchant.Dict("en")
>>> d.check("test")
True
>>> d.check(chr(2**16))
False
```

**$ python2**
```
Python 2.7.6 (default, Jun 22 2015, 17:58:13) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import enchant
>>> d = enchant.Dict("en")
>>> d.check("test")
True
>>> d.check(unichr(2**16).encode('utf8'))
False

```

### CODE REVIEW

**Comment by rfk** [comment]
Hi @halfak, if you're still interested in this, I think we should try to get it merged as a myspell-backend-specific fix.  I'm thinking something like:

* Create an `EnchantStr` subclass `UTF16EnchantStr` to encapsulate the extra logic here
* In the `enchant.Dict._switch_this` method, after reading the provider, check whether it's MySpell and set a local attribute to to either `EnchantStr` or `UTF16EnchantStr` as appropriate
* In the `check`, `suggest`, `add`, etc methods on `enchant.Dict`, use this local attribute rather than the default `EnchantStr` class.

What do you think?

**Comment by halfak** [comment]
This is a good proposal.  I've been too busy to pick it up recently and my workload doesn't look to by dying down soon.  Just posting to make sure you know I'm not ignoring you.  :) Thanks for being so responsive, @rtk.

File: enchant/__init__.py [hunk 1]
```diff
@@ -592,11 +592,16 @@ def _switch_this(self,this,broker):
         desc = self.__describe(check_this=False)
         self.tag = desc[0]
         self.provider = ProviderDesc(*desc[1:])
+        if self.provider.name == "myspell":
```
Comment (halfak) [inline enchant/__init__.py]: Here's where the magic happens.  OK to match all of 'myspell'?

**Comment by halfak** [comment]
@rtk, finally got a chance to do this.  What do you think?

**Comment by rfk** [comment]
Thanks @halfak!  Looks good to me, merging

**Comment by halfak** [comment]
\o/

**Pull request merged**